code/_onclick/click.dm

```diff
@@ -495,3 +495,17 @@
 			return TRUE
 
 	return FALSE
+
+/mob/proc/canGhostRole(role, use_cooldown = FALSE)
```

Comment:
This should be a client proc actually since it only uses values from client

Reply:
:+1:

Will get it in a few hours

Reply:
Done, updated title, description and briefly re-tested to ensure checks were being performed correctly. 

I had to slightly change the proc so that whatever is being treated as a spawner can pass along its flags_1 variable.